lib/maths/unittest/CTimeSeriesModelTest.cc

```diff
@@ -2330,6 +2330,106 @@ void CTimeSeriesModelTest::testDaylightSaving() {
     }
 }
 
+void CTimeSeriesModelTest::testSkipAnomalyModelUpdate() {
+    // We test that when parameters dictate to skip updating the anomaly model
+    // probabilities become smaller despite seeing many anomalies.
+
+    test::CRandomNumbers rng;
+
+    std::size_t length = 2000;
+    core_t::TTime bucketLength{600};
+
+    LOG_DEBUG(<< "Univariate");
+    {
+        TDoubleVec samples;
+        rng.generateNormalSamples(10.0, 2.0, length, samples);
+
+        maths::CTimeSeriesDecomposition trend{24.0 * DECAY_RATE, bucketLength};
+        maths::CUnivariateTimeSeriesModel model{modelParams(bucketLength), 1,
+                                                trend, univariateNormal()};
+
+        TDoubleVec probabilities;
+        TDouble2VecWeightsAryVec weights{maths_t::CUnitWeights::unit<TDouble2Vec>(1)};
+        core_t::TTime time{0};
+        for (std::size_t bucket = 0; bucket < samples.size(); ++bucket) {
+            auto sample = samples[bucket];
+            auto currentComputeProbabilityParams = computeProbabilityParams(weights[0]);
+
+            // We create anomalies in 10 consecutive buckets
+            if (bucket >= 1700 && bucket < 1710) {
+                sample = 100.0;
+                currentComputeProbabilityParams.skipAnomalyModelUpdate(true);
+            } else {
+                model.addSamples(addSampleParams(weights),
+                                 {core::make_triple(time, TDouble2Vec{sample}, TAG)});
+            }
+            maths::SModelProbabilityResult result;
+            model.probability(currentComputeProbabilityParams, {{time}}, {{sample}}, result);
+            if (result.s_Probability < 1e-6) {
```

Comment:
Tying this to probability seems a bit fragile to me. Could you perhaps just move
```
maths::SModelProbabilityResult result;
model.probability(currentComputeProbabilityParams, {{time}}, {{sample}}, result);
probabilities.push_back(result.s_Probability);
```
inside the `if (bucket >= 1700 && bucket < 1710)` case?